### PR TITLE
 feat: add if*generation*match args into Bucket.delete_blobs()

### PR DESCRIPTION
**Pull request by IlyaFaer**
Towards #127

### CODE REVIEW

**Comment by IlyaFaer** [comment]
This should be merged only after merging #123, as this branch based on the one from PR 123.

File: google/cloud/storage/bucket.py [hunk 1]
```diff
@@ -1227,15 +1367,53 @@ def delete_blobs(self, blobs, on_error=None, client=None, timeout=_DEFAULT_TIMEO
...
-                self.delete_blob(blob_name, client=client, timeout=timeout)
+                self.delete_blob(
+                    blob_name,
+                    client=client,
+                    timeout=timeout,
+                    if_generation_match=next(if_generation_match, None),
+                    if_generation_not_match=next(if_generation_not_match, None),
+                    if_metageneration_match=next(if_metageneration_match, None),
+                    if_metageneration_not_match=next(if_metageneration_not_match, None),
+                )
```
Comment (IlyaFaer) [inline google/cloud/storage/bucket.py:1416]: I this method we're converting args into iterators to be able to take next item (or `None` if there is no next item) with single line.

**Comment by IlyaFaer** [comment]
I don't see any system tests for this method. Maybe it should be added to check generation match args are working fine?

**Comment by andrewsg** [comment]
Since #123 has been merged, I be ready to review this once the branch is updated and conflicts resolved. Thanks!

**Comment by IlyaFaer** [comment]
@andrewsg, I've pushed some changes to check if generation match args length differs from `blobs` list (as we done in `compose()`) as well.

File: google/cloud/storage/bucket.py [hunk 2]
```diff
@@ -1398,15 +1408,74 @@ def delete_blobs(self, blobs, on_error=None, client=None, timeout=_DEFAULT_TIMEO
...
+            >>> from google.cloud import storage
+
+            >>> client = storage.Client()
+            >>> bucket = client.bucket("bucket-name")
+
+            >>> blobs = [bucket.blob("blob-name-1"), bucket.blob("blob-name-2")]
+            >>> if_generation_match = [None] * len(blobs)
+            >>> if_generation_match[0] = "123"  # precondition for "blob-name-1"
+
+            >>> bucket.delete_blobs(blobs, if_generation_match)
```
[the remark this answers is not in the document]
Comment (IlyaFaer) [reply google/cloud/storage/bucket.py]: Oh, yeah! Pushed!

**Pull request merged**